File: app/src/main/java/com/example/installer/utils/NetWorkUtil.java
package com.example.installer.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import com.example.installer.App;


public class NetWorkUtil {


    /**
     * 检测网络是否连接
     *
     * @return
     */
    public static boolean isNetConnected() {
        ConnectivityManager cm = (ConnectivityManager) App.instance.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
        if (cm != null) {
            NetworkInfo[] infos = cm.getAllNetworkInfo();
            if (infos != null) {
                for (NetworkInfo ni : infos) {
                    if (ni.isConnected()) {
                        return true;
                    }
                }
            }
        }
        return false;
    }

    /**
     * 检测wifi是否连接
     *
     * @return
     */
    public boolean isWifiConnected() {
        ConnectivityManager cm = (ConnectivityManager) App.instance.getApplicationContext()
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        if (cm != null) {
            NetworkInfo networkInfo = cm.getActiveNetworkInfo();
            if (networkInfo != null
                    && networkInfo.getType() == ConnectivityManager.TYPE_WIFI) {
                return true;
            }
        }
        return false;
    }

    /**
     * 检测3G是否连接
     *
     * @return
     */
    public boolean is3gConnected() {
        ConnectivityManager cm = (ConnectivityManager) App.instance.getApplicationContext()
                .getSystemService(Context.CONNECTIVITY_SERVICE);
        if (cm != null) {
            NetworkInfo networkInfo = cm.getActiveNetworkInfo();
            if (networkInfo != null
                    && networkInfo.getType() == ConnectivityManager.TYPE_MOBILE) {
                return true;
            }
        }
        return false;
    }

}

File: app/src/main/java/com/example/installer/utils/NetWorkUtil.kt
package com.example.installer.utils

import android.content.Context
import android.net.ConnectivityManager
import com.example.installer.App

/**
@author zhangtao
@date   2020/11/3

 **/
class NetWorkUtil {
    companion object{

        /**
         * 检测网络是否连接
         *
         * @return
         */
        fun isNetConnected(): Boolean {
            val cm =
                App.instance.applicationContext.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            if (cm != null) {
                val infoList = cm.allNetworkInfo
                if (infoList != null) {
                    for (ni in infoList) {
                        if (ni.isConnected) {
                            return true
                        }
                    }
                }
            }
            return false
        }

        /**
         * 检测wifi是否连接
         *
         * @return
         */
        fun isWifiConnected(): Boolean {
            val cm = App.instance.applicationContext
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            if (cm != null) {
                val networkInfo = cm.activeNetworkInfo
                if (networkInfo != null
                    && networkInfo.type == ConnectivityManager.TYPE_WIFI
                ) {
                    return true
                }
            }
            return false
        }

        /**
         * 检测3G是否连接
         *
         * @return
         */
        fun is3gConnected(): Boolean {
            val cm = App.instance.applicationContext
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            if (cm != null) {
                val networkInfo = cm.activeNetworkInfo
                if (networkInfo != null
                    && networkInfo.type == ConnectivityManager.TYPE_MOBILE
                ) {
                    return true
                }
            }
            return false
        }

    }
}